src/frontends/lean/tactic_notation.cpp

```diff
@@ -156,34 +160,49 @@ static expr parse_interactive_param(parser & p, expr const & ty, expr const & qu
-    name irtactic(name(tac_class, "interactive"), "irtactic");
+    expr type     = p.env().get(decl_name).get_type();
+    name itactic  = get_interactive_tactic_full_name(tac_class, "itactic");
+    name irtactic = get_interactive_tactic_full_name(tac_class, "irtactic");
     buffer<expr> args;
-    while (is_pi(type)) {
-        if (is_explicit(binding_info(type))) {
-            expr arg_type = binding_domain(type);
-            if (is_app_of(arg_type, get_interactive_parse_name())) {
-                buffer<expr> arg_args;
-                get_app_args(arg_type, arg_args);
-                lean_assert(arg_args.size() == 3);
-                args.push_back(parse_interactive_param(p, arg_args[0], arg_args[1], arg_args[2]));
-            } else if (is_constant(arg_type, itactic)) {
-                bool report_error = false;
-                args.push_back(parse_nested_auto_quote_tactic(p, tac_class, use_rstep, report_error));
-            } else if (is_constant(arg_type, irtactic)) {
-                args.push_back(parse_nested_auto_quote_tactic(p, tac_class, use_rstep, report_error));
-            } else {
-                break;
+    try {
+        try {
+            p.next();
+            while (is_pi(type)) {
+                p.check_break_before();
+                if (is_explicit(binding_info(type))) {
+                    expr arg_type = binding_domain(type);
+                    if (is_app_of(arg_type, get_interactive_parse_name())) {
+                        buffer<expr> arg_args;
+                        get_app_args(arg_type, arg_args);
+                        lean_assert(arg_args.size() == 3);
+                        args.push_back(parse_interactive_param(p, arg_args[0], arg_args[1], arg_args[2]));
+                    } else if (is_constant(arg_type, itactic)) {
+                        bool report_error = false;
+                        args.push_back(parse_nested_interactive_tactic(p, tac_class, use_rstep, report_error));
+                    } else if (is_constant(arg_type, irtactic)) {
+                        args.push_back(parse_nested_interactive_tactic(p, tac_class, use_rstep, report_error));
+                    } else {
+                        break;
+                    }
+                }
+                type = binding_body(type);
+            }
+            while (p.curr_lbp() >= get_max_prec()) {
+                p.check_break_before();
+                args.push_back(p.parse_expr(get_max_prec()));
             }
+            p.check_break_before();
+        } catch (break_at_pos_exception) {
+            throw;
+        } catch (...) {
+            p.check_break_before();
+            throw;
         }
-        type = binding_body(type);
-    }
-    while (p.curr_lbp() >= get_max_prec()) {
-        args.push_back(p.parse_expr(get_max_prec()));
+    } catch (break_at_pos_exception & e) {
+        e.m_token_info.m_tac_param_idx = args.size();
+        throw;
```

Comment:
The `break_at_pos_exception` handling is slowly getting ridiculous, but when(/if?) we switch to embedding info in server-generated highlight instead, I'll have to redo it anyway.